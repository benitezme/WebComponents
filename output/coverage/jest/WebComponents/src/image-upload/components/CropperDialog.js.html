<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebComponents/src/image-upload/components/CropperDialog.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">WebComponents/src/image-upload/components</a> CropperDialog.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React from 'react'</span>
<span class="cstat-no" title="statement not covered" >import PropTypes from 'prop-types'</span>
<span class="cstat-no" title="statement not covered" >import { withStyles } from '@material-ui/core/styles'</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >import Dialog from '@material-ui/core/Dialog'</span>
<span class="cstat-no" title="statement not covered" >import DialogActions from '@material-ui/core/DialogActions'</span>
<span class="cstat-no" title="statement not covered" >import DialogContent from '@material-ui/core/DialogContent'</span>
<span class="cstat-no" title="statement not covered" >import Button from '@material-ui/core/Button'</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >import { Cropper } from 'react-image-cropper'</span>
&nbsp;
const <span class="fstat-no" title="function not covered" >styles = <span class="cstat-no" title="statement not covered" >t</span>heme<span class="cstat-no" title="statement not covered" > =&gt; ({</span></span>
  dropzone: {
    display: 'block',
    padding: '3em',
    height: 200,
    borderWidth: 2,
    borderColor: '#999',
    backgroundColor: '#eee'
  }
})
&nbsp;
export const <span class="fstat-no" title="function not covered" >CropperDialog = <span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" >(</span>{ classes, open, handleClose, imgSrc, handleCrop, handleCancel, cropperRef }) =&gt; (</span></span></span>
  &lt;Dialog
    open={open}
    onClose={handleClose}
    aria-labelledby='form-dialog-title'
  &gt;
    &lt;div classes={classes.dialogContainer}&gt;
      &lt;DialogContent&gt;
        &lt;div className={classes.cropImageBox} &gt;
          &lt;Cropper
            src={imgSrc}
            originX={100}
            originY={100}
            ref={cropperRef}
            ratio={1}
          /&gt;
        &lt;/div&gt;
      &lt;/DialogContent&gt;
      &lt;DialogActions&gt;
        &lt;Button onClick={handleCancel} color='primary'&gt;
          Cancel
        &lt;/Button&gt;
        &lt;Button
          onClick={handleCrop}
          color='primary'
        &gt;
          Crop Image
        &lt;/Button&gt;
      &lt;/DialogActions&gt;
    &lt;/div&gt;
  &lt;/Dialog&gt;
)
&nbsp;
<span class="cstat-no" title="statement not covered" >CropperDialog.propTypes = {</span>
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  imgSrc: PropTypes.object.isRequired,
  handleCrop: PropTypes.func.isRequired,
  cropperRef: PropTypes.node.isRequired,
  handleCancel: PropTypes.func.isRequired
}
&nbsp;
export default withStyles(styles)(CropperDialog)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 02 2018 01:24:37 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
