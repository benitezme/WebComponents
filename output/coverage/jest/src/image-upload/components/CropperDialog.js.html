<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/image-upload/components/CropperDialog.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/image-upload/components</a> CropperDialog.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
&nbsp;
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import Button from '@material-ui/core/Button'
&nbsp;
import { Cropper } from 'react-image-cropper'
&nbsp;
const styles = theme =&gt; ({
  dropzone: {
    display: 'block',
    padding: '3em',
    height: 200,
    borderWidth: 2,
    borderColor: '#999',
    backgroundColor: '#eee'
  }
})
&nbsp;
export const CropperDialog = ({ classes, open, handleClose, imgSrc, handleCrop, handleCancel, cropperRef }) =&gt; (
  &lt;Dialog
    open={open}
    onClose={handleClose}
    aria-labelledby='form-dialog-title'
  &gt;
    &lt;div classes={classes.dialogContainer}&gt;
      &lt;DialogContent&gt;
        &lt;div className={classes.cropImageBox} &gt;
          &lt;Cropper
            src={imgSrc}
            originX={100}
            originY={100}
            ref={cropperRef}
            ratio={1}
          /&gt;
        &lt;/div&gt;
      &lt;/DialogContent&gt;
      &lt;DialogActions&gt;
        &lt;Button onClick={handleCancel} color='primary'&gt;
          Cancel
        &lt;/Button&gt;
        &lt;Button
          onClick={handleCrop}
          color='primary'
        &gt;
          Crop Image
        &lt;/Button&gt;
      &lt;/DialogActions&gt;
    &lt;/div&gt;
  &lt;/Dialog&gt;
)
&nbsp;
CropperDialog.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  imgSrc: PropTypes.string.isRequired,
  handleCrop: PropTypes.func.isRequired,
  cropperRef: PropTypes.node.isRequired,
  handleCancel: PropTypes.func.isRequired
}
&nbsp;
export default withStyles(styles)(CropperDialog)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 02 2018 01:51:43 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
