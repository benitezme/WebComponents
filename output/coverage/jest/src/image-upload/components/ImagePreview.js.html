<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/image-upload/components/ImagePreview.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/image-upload/components</a> ImagePreview.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
&nbsp;
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
&nbsp;
const styles = <span class="fstat-no" title="function not covered" >th</span>eme =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  imagePreview: {
    width: 100,
    height: 100
  }
})
&nbsp;
export const ImagePreview = <span class="fstat-no" title="function not covered" >({</span> classes, handleChangeImage, image }) =&gt; (
<span class="cstat-no" title="statement not covered" >  &lt;React.Fragment&gt;</span>
    &lt;img src={image} className={classes.imagePreview} alt='Existing Image Preview' /&gt;
    &lt;Button
      onClick={handleChangeImage}
      color='primary'
    &gt;
      Change
    &lt;/Button&gt;
  &lt;/React.Fragment&gt;
)
&nbsp;
ImagePreview.propTypes = {
  classes: PropTypes.object.isRequired,
  handleChangeImage: PropTypes.func,
  image: PropTypes.string
}
&nbsp;
export default withStyles(styles)(ImagePreview)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 02 2018 01:51:43 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
