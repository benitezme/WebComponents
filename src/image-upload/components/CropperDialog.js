import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import Button from '@material-ui/core/Button'

import { Cropper } from 'react-image-cropper'

const styles = theme => ({
  dropzone: {
    display: 'block',
    padding: '3em',
    height: 200,
    borderWidth: 2,
    borderColor: '#999',
    backgroundColor: '#eee'
  }
})

class CropperDialog extends Component {
  render () {
    const { classes, open, handleClose, imgSrc, handleCrop, handleCancel, cropperRef } = this.props
    console.log(cropperRef)
    return (
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='form-dialog-title'
      >
        <div classes={classes.dialogContainer}>
          <DialogContent>
            <div className={classes.cropImageBox} >
              {imgSrc === null && (
                <div>Loading...</div>
              )}
              {imgSrc !== null && (
                <Cropper
                  src={imgSrc}
                  originX={100}
                  originY={100}
                  ref={ref => { this.cropper = ref }}
                  ratio={1}
                />
              )}
            </div>
          </DialogContent>
          <DialogActions>
            <Button onClick={handleCancel} color='primary'>
              Cancel
            </Button>
            <Button
              onClick={(e) => handleCrop(e, this.cropper)}
              color='primary'
            >
              Crop Image
            </Button>
          </DialogActions>
        </div>
      </Dialog>
    )
  }
}

CropperDialog.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  imgSrc: PropTypes.string,
  handleCrop: PropTypes.func.isRequired,
  cropperRef: PropTypes.any.isRequired,
  handleCancel: PropTypes.func.isRequired
}

export default withStyles(styles)(CropperDialog)
