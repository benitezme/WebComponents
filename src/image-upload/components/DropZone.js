import React from 'react'
import PropTypes from 'prop-types'

import Dropzone from 'react-dropzone'

import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const styles = theme => ({
  dropzone: {
    display: 'block',
    padding: '3em',
    height: 200,
    borderWidth: 2,
    borderColor: '#999',
    backgroundColor: '#eee'
  },
  dropzoneText: {
    textAlign: 'center'
  }
})

export const DropZone = ({ classes, onDrop, handleCancel }) => (
  <React.Fragment>
    <Dropzone
      accept='image/jpeg, image/png'
      onDrop={(e) => onDrop(e)}
      className={classes.dropzone}
      multiple={false}
    >
      <p className={classes.dropzoneText}>Drag-and-drop image here or click to select <br /> Only *.jpeg and *.png images</p>
    </Dropzone>
    <Button
      onClick={handleCancel}
      color='primary'
    >
      Cancel Image Change
    </Button>
  </React.Fragment>
)

DropZone.propTypes = {
  classes: PropTypes.object.isRequired,
  onDrop: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired
}

export default withStyles(styles)(DropZone)
