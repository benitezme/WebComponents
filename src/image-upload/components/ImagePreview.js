import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const styles = theme => ({
  imagePreview: {
    width: 100,
    height: 100
  }
})

export const ImagePreview = ({ classes, handleChangeImage, image }) => (
  <React.Fragment>
    <img src={image} className={classes.imagePreview} alt='Existing Image Preview' />
    <Button
      onClick={handleChangeImage}
      color='primary'
    >
      Change
    </Button>
  </React.Fragment>
)

ImagePreview.propTypes = {
  classes: PropTypes.object.isRequired,
  handleChangeImage: PropTypes.func,
  image: PropTypes.string
}

export default withStyles(styles)(ImagePreview)
