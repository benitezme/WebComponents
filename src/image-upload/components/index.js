import DropZone from './DropZone'
import ImagePreview from './ImagePreview'
import CropperDialog from './CropperDialog'

export { DropZone, ImagePreview, CropperDialog }
