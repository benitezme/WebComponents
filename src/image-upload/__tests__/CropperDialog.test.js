import React from 'react'
import renderer from 'react-test-renderer'
import { unwrap } from '@material-ui/core/test-utils'

import { shallow } from '../../enzyme'

import { CropperDialog } from '../components'

let handleCloseMock = jest.fn()
let handleCropMock = jest.fn()
let handleCancelMock = jest.fn()
const cropPreview = {
  width: 350,
  height: 350
}
const placeHolder =
  'https://aadevelop.blob.core.windows.net/module-web-components/assets/Superalgos-logo-horz-sm.jpg'

const ComponentNaked = unwrap(CropperDialog)

describe('CropperDialog tests', () => {
  it('with shallow', () => {
    const wrapper = shallow(
      <ComponentNaked
        open={false}
        imgSrc={placeHolder}
        handleClose={handleCloseMock}
        handleCrop={handleCropMock}
        handleCancel={handleCancelMock}
        cropImagePreview={cropPreview}
        cropperRef={e => this.instance()}
        cropRatio={1}
        classes={{
          dialogContainer: '',
          cropImageBox: ''
        }}
      />
    )
    console.log('shallow', wrapper.debug())
    expect(wrapper.find('CropperDialog')).toBeDefined()
  })

  it('renders matching snapshot with dialog closed', () => {
    const tree = renderer
      .create(
        <ComponentNaked
          open={false}
          imgSrc={placeHolder}
          handleClose={handleCloseMock}
          handleCrop={handleCropMock}
          handleCancel={handleCancelMock}
          cropImagePreview={cropPreview}
          cropperRef={e => this.instance()}
          cropRatio={1}
          classes={{
            dialogContainer: '',
            cropImageBox: ''
          }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders matching snapshot with dialog open', () => {
    const tree = renderer
      .create(
        <ComponentNaked
          open={true}
          imgSrc={placeHolder}
          handleClose={handleCloseMock}
          handleCrop={handleCropMock}
          handleCancel={handleCancelMock}
          cropImagePreview={cropPreview}
          cropperRef={e => this.instance()}
          cropRatio={1}
          classes={{
            dialogContainer: '',
            cropImageBox: ''
          }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders', () => {
    const wrapper = shallow(
      <CropperDialog
        open={false}
        imgSrc={placeHolder}
        handleClose={handleCloseMock}
        handleCrop={handleCropMock}
        handleCancel={handleCancelMock}
        cropImagePreview={cropPreview}
        cropperRef={e => this.instance()}
        cropRatio={1}
        classes={{
          dialogContainer: '',
          cropImageBox: ''
        }}
      />
    )

    // Expect the wrapper object to be defined
    expect(wrapper.find('CropperDialog')).toBeDefined()
  })
})
