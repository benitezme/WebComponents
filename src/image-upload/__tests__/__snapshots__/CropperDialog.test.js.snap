// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard tests renders matching snapshot with dialog closed 1`] = `null`;
